url64700e9ab7_dwicke_course_scheduler_tgz-pJ8-org_json_TestJ8/src/coursescheduleapp/view/ResultsView.java

```diff
@@ -0,0 +1,299 @@
+public class ResultsView extends javax.swing.JPanel implements Observer, ActionListener, HyperlinkListener {
+
+	private Model model;
+	private ResultsController control;
+	private ArrayList<JButton> removeBt;
+	private ArrayList<Course> courseResults;
+	private SchedViewI schedViewChoice;
+
+	/** Creates new form ResultsView */
+	public ResultsView() {
+		initComponents();
+	}
+
+	public void setModel(Model model)
+	{
+		this.model = model;
+		this.model.addObserver(this);
+		removeBt = new ArrayList<JButton>();
+		control = new ResultsController();
+		schedViewChoice = new OrderedSchedView();
+	}
+
+	/** This method is called from within the constructor to
+	 * initialize the form.
+	 * WARNING: Do NOT modify this code. The content of this method is
+	 * always regenerated by the Form Editor.
+	 */
+	@SuppressWarnings("unchecked")
+	// <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
+	private void initComponents() {
+
+		jScrollPane1 = new javax.swing.JScrollPane();
+		jTextPane = new javax.swing.JTextPane();
+		jTextPane.setEditable(false);
+		jTextPane.setEditorKit(new javax.swing.text.html.HTMLEditorKit());
+		jTextPane.setDocument(new javax.swing.text.html.HTMLDocument());
+		
+		
+		jTextPane.addHyperlinkListener(this);
+
+		printTxtPane = new javax.swing.JTextPane();
+		printTxtPane.setContentType("text/html");
+
+		setLayout(new javax.swing.BoxLayout(this, javax.swing.BoxLayout.LINE_AXIS));
+		jTextPane.setContentType("text/html");
+		
+		jTextPane.setStyledDocument(new javax.swing.text.html.HTMLDocument());
+		
+
+		jScrollPane1.setViewportView(jTextPane);
+
+
+		add(jScrollPane1);
+
+		fileChooser = new javax.swing.JFileChooser();
+	}// </editor-fold>//GEN-END:initComponents
+
+
+	// Variables declaration - do not modify//GEN-BEGIN:variables
+	private javax.swing.JScrollPane jScrollPane1;
+	// Need PrintTxtPane in order to print other wise embedded buttons will disappear
+	private javax.swing.JTextPane jTextPane, printTxtPane;
+	private javax.swing.JFileChooser fileChooser;
+
+	// End of variables declaration//GEN-END:variables
+	@Override
+	public void update(Observable o, Object arg) {
+		// TODO Auto-generated method stub
+		Schedule sched = model.getBestSchedule();
+
+		jTextPane.setText("");
+		printTxtPane.setText("");
+		removeBt.clear();
+
+		for (int i = 0; i < removeBt.size(); i++)
+		{
+			jTextPane.remove(removeBt.get(i));
+			System.out.println("Button Removed");
+		}
+		jTextPane.validate();
+
+
+
+		Iterator<Course> it = sched.getCourses().iterator();
+
+		if (it.hasNext() == false)
+		{
+			try {
+				jTextPane.getStyledDocument().insertString(jTextPane.getStyledDocument().getLength(), "Sorry, no schedule can be created.", null);
+				printTxtPane.getStyledDocument().insertString(printTxtPane.getStyledDocument().getLength(), "Sorry, no schedule can be created.", null);
+			} catch (BadLocationException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 141:147, 168:175, 189:196.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.